src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -186,6 +195,9 @@ private String serializeBatch(final Partition.PartitionKey partitionKey, final S
             events.stream().forEach(event -> builder.append(event).append(","));
             builder.deleteCharAt(builder.length() - 1).append("]");
         }
+        if (metadata.isPresent()) {
```

Comment:
For me this one looks cleaner:
metadata.ifPresent(data->builder.append(",\"metadata\":{\"debug\":\"").append(data).append("\"}"))